src/microbe_stage/editor/MicrobePartSelection.cs

```diff
@@ -136,17 +137,30 @@ public override void _ExitTree()
 
     private void OnDisplayPartNamesChanged(bool displayed)
     {
-        if (nameLabel != null)
-            nameLabel.Visible = displayed || AlwaysShowLabel;
+        if (nameLabel == null)
+            return;
+
+        var flag = displayed || AlwaysShowLabel;
+
+        nameLabel.Visible = flag;
+
+        var vbox = GetChild<Control>(0);
+        if (flag)
+        {
+            vbox.AddConstantOverride("separation", 1);
+        }
+        else
+        {
+            vbox.AddConstantOverride("separation", 4);
+        }
     }
 
     private void UpdateLabels()
     {
         if (mpLabel == null || nameLabel == null)
             return;
 
-        mpLabel.Text = TranslationServer.Translate("MP_COST").FormatSafe(MPCost);
```

Comment:
When investigating this change I noticed that there are multiple other places still in the game where the MP cost is not shown with the icon...